src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts

```diff
@@ -128,7 +130,7 @@ function CustomRouter(options: ResponsiveStackNavigatorRouterOptions) {
             return state;
         },
         getStateForAction(state: StackNavigationState<ParamListBase>, action: CommonActions.Action | StackActionType, configOptions: RouterConfigOptions) {
-            if (shouldPreventReset(state, action)) {
+            if (shouldPreventReset(state, action) && isAuthenticated) {
```

Comment:
Can we move checking `isAuthenticated` to the `shouldPreventReset` and make an early return if it's false?